Cabal/Distribution/PackageDescription/Configuration.hs

```diff
@@ -517,38 +518,25 @@ flattenPackageDescription
         , executables  = reverse exes
         , testSuites   = reverse tests
         , benchmarks   = reverse bms
-        , buildDepends = ldeps
-                      ++ reverse sub_ldeps
-                      ++ reverse pldeps
-                      ++ reverse edeps
-                      ++ reverse tdeps
-                      ++ reverse bdeps
         }
   where
-    (mlib, ldeps) = case mlib0 of
-        Just lib -> let (l,ds) = ignoreConditions lib in
-                    (Just ((libFillInDefaults l) { libName = Nothing }), ds)
-        Nothing -> (Nothing, [])
-    (sub_libs, sub_ldeps) = foldr flattenLib  ([],[]) sub_libs0
-    (flibs,    pldeps)    = foldr flattenFLib ([],[]) flibs0
-    (exes,     edeps)     = foldr flattenExe  ([],[]) exes0
-    (tests,    tdeps)     = foldr flattenTst  ([],[]) tests0
-    (bms,      bdeps)     = foldr flattenBm   ([],[]) bms0
-    flattenLib (n, t) (es, ds) =
-        let (e, ds') = ignoreConditions t in
-        ( (libFillInDefaults $ e { libName = Just n, libExposed = False }) : es, ds' ++ ds )
-    flattenFLib (n, t) (es, ds) =
-        let (e, ds') = ignoreConditions t in
-        ( (flibFillInDefaults $ e { foreignLibName = n }) : es, ds' ++ ds )
-    flattenExe (n, t) (es, ds) =
-        let (e, ds') = ignoreConditions t in
-        ( (exeFillInDefaults $ e { exeName = n }) : es, ds' ++ ds )
-    flattenTst (n, t) (es, ds) =
-        let (e, ds') = ignoreConditions t in
-        ( (testFillInDefaults $ e { testName = n }) : es, ds' ++ ds )
-    flattenBm (n, t) (es, ds) =
-        let (e, ds') = ignoreConditions t in
-        ( (benchFillInDefaults $ e { benchmarkName = n }) : es, ds' ++ ds )
+    mlib = f <$> mlib0
+      where f lib = (libFillInDefaults . fst . ignoreConditions $ lib) { libName = Nothing }
+    sub_libs = flattenLib  <$> sub_libs0
+    flibs    = flattenFLib <$> flibs0
+    exes     = flattenExe  <$> exes0
+    tests    = flattenTst  <$> tests0
+    bms      = flattenBm   <$> bms0
+    flattenLib (n, t) = libFillInDefaults $ (fst $ ignoreConditions t)
+      { libName = Just n, libExposed = False }
+    flattenFLib (n, t) = flibFillInDefaults $ (fst $ ignoreConditions t)
+      { foreignLibName = n }
+    flattenExe (n, t) = exeFillInDefaults $ (fst $ ignoreConditions t)
+      { exeName = n }
+    flattenTst (n, t) = testFillInDefaults $ (fst $ ignoreConditions t)
+      { testName = n }
+    flattenBm (n, t) = benchFillInDefaults $ (fst $ ignoreConditions t)
+      { benchmarkName = n }
```

Comment:
What happened in this diff?

Reply:
I removed the accumulation of deps and just did the flattening of components. Because of the removal of https://github.com/haskell/cabal/pull/4383/files#diff-12c8f4f2a1b901f8cbc37a619afafcb7L520, we no longer need that accumulation.